main.go

```diff
@@ -17,25 +17,34 @@ type conf struct {
 	Token string `yaml:"token"`
 }
 
-func (c *conf) getConf() *conf {
-
+func (c *conf) getConf() (*conf, error) {
 	yamlFile, err := ioutil.ReadFile("config.yaml")
 	if err != nil {
-		log.Printf("yamlFile.Get err   #%v ", err)
+		return nil, fmt.Errorf("failed to read configuration file: %v", err)
 	}
+
 	err = yaml.Unmarshal(yamlFile, c)
 	if err != nil {
-		log.Fatalf("Unmarshal: %v", err)
+		return nil, fmt.Errorf("failed to unmarshall configuration file into yaml: %v", err)
 	}
 
-	return c
+	return c, nil
```

Comment:
@jaredallard Should the return statement be 'return c, err' since err will be nil and this way, it will improve code readability?

Reply:
No, because we're doing an explicit `err != nil` check above, we're doing that to better format the error message that gets returned, if we weren't doing that then we'd return err.